### PR TITLE
support arg unpacking

### PR DESCRIPTION
**Pull request by tjhance**
Support arg unpacking

This doesn't work for single-element tuples, due to a bug in pypa. (I submitted a PR to dropbox/libpypa to fix this.)

### CODE REVIEW

**Comment by tjhance** [comment]
I also got the `global_and_local.py` test to pass, which checks that we throw an exception when you try to declare a param name as global.

(A bit unrelated, but I put it in the same PR to avoid conflicts.)

**Comment by tjhance** [comment]
I updated the libpypa submodule and added a test case for the single-element tuple argument.

**Comment by undingen** [comment]
I think BeautifulSoup.py depends on this change because it uses:
```python
convert = lambda(k, val): (k, re.sub("&(#\d+|#x[0-9a-fA-F]+|\w+);",
                                           self._convertEntities,
                                           val))

```
Whats the status on this pull request?

**Comment by tjhance** [comment]
Ready as far as I know, it just needs review.

**Comment by tjhance** [comment]
Added arg unpacking support for lambdas.

File: test/tests/global_and_local.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
-# expected: fail
 # - this particular check isn't implemented yet
```
Comment (kmod) [inline test/tests/global_and_local.py]: we should probably remove this comment too?

File: src/core/cfg.cpp [hunk 2]
```diff
@@ -2502,6 +2502,32 @@ CFG* computeCFG(SourceInfo* source, std::vector<AST_stmt*> body) {
...
+            args = ast_cast<AST_Lambda>(source->ast)->args;
+        }
+        int counter = 0;
+        for (AST_expr* arg_expr : args->args) {
+            if (arg_expr->type == AST_TYPE::Tuple) {
+                InternedString arg_name = source->getInternedStrings().get("." + std::to_string(counter));
+                AST_Name* arg_name_expr
+                    = new AST_Name(arg_name, AST_TYPE::Load, arg_expr->lineno, arg_expr->col_offset);
+                visitor.pushAssign(arg_expr, arg_name_expr);
+            }
```
Comment (kmod) [inline src/core/cfg.cpp]: are Tuples the only things that can get this treatment?  a quick test seems to indicate it, but an assert that that the type is a tuple or a name could make that clear.

**Comment by kmod** [comment]
lgtm

**Comment by tjhance** [comment]
@kmod implemented your comments

**Pull request merged**